static/src/javascripts/projects/commercial/modules/check-dispatcher.js

```diff
@@ -69,24 +73,30 @@ const checksToDispatch = {
         // if thirdPartyTags false no external ads are loaded
         // is irrelevant for ad-free users (independently of thirdPartyTags)
         if (commercialFeatures.thirdPartyTags && !commercialFeatures.adFree) {
-            return waitForCheck('hasHighPriorityAdLoaded').then(
-                highPriorityAdLoaded => {
+            return waitForCheck('hasHighPriorityAdLoaded')
```

Comment:
Can you not just add the `catch` to the Promise returned from `waitForCheck`?

Reply:
@shtukas yes you could do as @gtrufitt suggested, put a single catch on the `Promise` that `waitForCheck` returns, which is here: https://github.com/guardian/frontend/blob/master/static/src/javascripts/projects/common/modules/check-mediator.js#L13

Then you don't have to catch each individual `waitForCheck` call.